### ISSUE TITLE
reproduce your experiment results

### ISSUE THREAD

**Issue by Shawn0002** [tudelft3d/Building-PCC-Building-Point-Cloud-Completion-Benchmarks#5]
I want to reproduce the point cloud completion figure in your paper, but I can't find any reference to help me reproduce your experimental results

**Comment by WeixiaoGao** [comment]
Can you be more specific? Which figure or experiment do you want to reproduce?

**Comment by Shawn0002** [comment]
I would like to use the provided pre-trained model to reproduce Figure 3 in the original paper, which is the result of the building point cloud completion visualization.

**Comment by Shawn0002** [comment]
I want to use the provided pre-trained model to reproduce Figure 3 in the original article, which is the visual result of the building point cloud completion. I want to know which specific point cloud in the dataset the six buildings in Figure 3 correspond to. I also want to ask you to help me know whether the provided pre-trained model is trained on the building dataset.

**Comment by Shawn0002** [comment]
I want to use the provided pre-trained model to reproduce Figure 3 in the original article, which is the visual result of the building point cloud completion. I want to know which specific point cloud in the dataset the six buildings in Figure 3 correspond to. I also want to ask you to help me know whether the provided pre-trained model is trained on the building dataset. Looking forward ur early reply！

**Comment by WeixiaoGao** [comment]
The IDs of these six buildings are: bag_0518100000340196, bag_0518100000340743, bag_0518100000342812, bag_0518100001637311, bag_0518100001764160, BAG_0599100010071777. The pre-trained model was trained on our building dataset.

**Comment by Shawn0002** [comment]
Thank you very much for your guidance! Your detailed and timely response has been incredibly helpful in my understanding and progress on this project. I really appreciate the effort you've put into assisting me, and I’m excited to try out your suggestions right away!

**Comment by Shawn0002** [comment]
Thank you very much for your guidance! Your detailed and timely response has been incredibly helpful in my understanding and progress on this project.

**Comment by Shawn0002** [comment]
When I used the pre-trained model you provided to reproduce the experimental results in your paper, I used the following command: 

**`python tools\inference.py --model_config .\cfgs\BuildingNL_models\AdaPoinTr.yaml --model_checkpoint .\pr_tr_mdoels\AdaPoinTr\BNL_50k_600e\ckpt-best.pth --pc_root ".\fig3" --save_vis_img False --out_pc_root ".\reproduce_results\inference_result_AdaPoinTr\" --save_merged True`**. 

The results I got were quite different from the experimental figures in your paper, and there was basically no point cloud on the side. I don't know if there was a problem with my operation. I would like to ask this question and look forward to your early reply.

**Comment by Shawn0002** [comment]
![image](https://github.com/user-attachments/assets/60ca7145-a43d-427e-be57-e79dfdd1b676)
I'll take Figure 3 in your paper as an example. The point cloud completion result of the fourth building is as follows，I want to know how to reproduce the experimental results in Figure 3 of the paper using the pre-trained model, but I did not get the expected results. I hope to get your guidance so that I can reproduce your experimental results. Thanks！！

**Comment by WeixiaoGao** [comment]
Please first check [this command example](https://github.com/tudelft3d/Building-PCC-Building-Point-Cloud-Completion-Benchmarks/blob/de6cfda5be7ce6508b3097548a4b3e8117b0c797/code/PoinTr/file_operation_scripts) before you begin to infer or reproduce our results. Let me know if you still cannot get it.

**Comment by Shawn0002** [comment]
Thanks for your guidance and help, after checking file_operation_files i found that  the 

> --truth_root

 parameter was not set in the code I originally ran, which was the code that could not get the expected point cloud completion results. After setting this parameter, I got the same results as expected. But I want to know why I need to input ground truth when testing the results? In short, thank you for your patience and guidance in helping me reproduce your experimental results.
![fig3_(4)](https://github.com/user-attachments/assets/42348f47-23dc-4a7c-a5dd-a4404040203e)

**Comment by Shawn0002** [comment]
Dear Maintainers,

I have been working on reproducing the code provided in this repository and noticed a potential issue regarding the use of the truth_root parameter during the inference stage. It appears that this parameter introduces prior knowledge, which could affect the model's inference results.

In the inference.py file, the normalization of input point clouds during the inference phase depends on whether the truth_root parameter is provided. Specifically, the following code snippet demonstrates this behavior:
`if args.truth_root != '':
    _, centroid, m = pc_norm(gt_ndarray)
    pc_ndarray_trans = pc_norm_with_centroid_and_scale(pc_ndarray, centroid, m)
else:
    pc_ndarray_trans, centroid, m = pc_norm(pc_ndarray)
`
When truth_root is specified, the normalization of the input point cloud (pc_ndarray) uses the centroid and scale (m) calculated from the ground truth point cloud (gt_ndarray). However, this introduces prior knowledge into the inference process, as the ground truth should not be accessible during inference in a real-world scenario. This could lead to a discrepancy between the model's actual performance in independent inference scenarios and the results observed when the truth_root parameter is used.
I want to know :
1、Was the use of truth_root during inference intentional, or is it a design oversight?
2、Should the inference process instead rely solely on the input point cloud (pc_ndarray) for normalization to ensure generalizability?
Thank you for your time and consideration! I look forward to your feedback on this matter.

**Comment by WeixiaoGao** [comment]
Here’s the corrected version:

Your observation is absolutely correct, and I have mentioned all these issues in the associated paper [Building-PCC: Building Point Cloud Completion Benchmarks](https://isprs-annals.copernicus.org/articles/X-4-W5-2024/179/2024/isprs-annals-X-4-W5-2024-179-2024.pdf). Please refer to Section 5.3. 

The short answers to your questions are as follows:
1. The `truth_root` is designed to provide the center and scale from ground truth data.  
2. Use the partial point cloud for normalization; the center and scale will deviate from the ground truth as current deep learning methods cannot infer them from a partial input.

The potential solutions are:  
1. Design a new network structure to predict the center and scale from a partial input.  
2. Use additional data, such as a 2D footprint and the partial input, to estimate the center and scale for the predicted data.  
3. Normalize all building models within a fixed range or scale during training and prediction, especially if your dataset contains buildings of similar sizes.

**Comment by ranhaogong** [comment]
@Shawn0002 Can I add your WeChat, I'm also working on this field and we're from the same school 😄

**Comment by Shawn0002** [comment]
19997186875 Wechat

Haoran Gong ***@***.***>于2024年12月26日 周四16:14写道：

**Comment by WeixiaoGao** [comment]
Thank you all for your interest. The repository is designed for a dataset; to reproduce the dataset itself, you might need an FME license. For testing the related deep learning frameworks, please use the code provided by us in this repository, as we have made some modifications to adapt it for this task. 

Please let me know if you have more questions. You can either raise an issue on GitHub or send me an email for personal questions. I am happy to answer them, but I cannot guarantee that I can do so in a timely manner.

**Comment by Shawn0002** [comment]
> > [@Shawn0002](https://github.com/Shawn0002)我可以加你微信吗，我也是做这个的，我们是同一个学校的😄
> 
> 你好，你成功复现了这个项目没？我最近也在研究基于点云重建建筑

我已经成功地复现了作者的这篇工作
